pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -81,8 +84,10 @@ func NewDynamicReconciler(rc pb.ResultsClient, lc pb.LogsClient, oc ObjectClient
 		resultsClient: results.NewClient(rc, lc),
 		objectClient:  oc,
 		cfg:           cfg,
-		// Always true predicate.
 		IsReadyForDeletionFunc: func(ctx context.Context, object results.Object) (bool, error) {
+			if LogsFinalizerExist(object, LogFinalizer) {
```

Comment:
I would think that this is not needed, as it is redundant with the `finalizer` pattern. 

I would expect the resource to be marked as being deleted, but the deletion to be defered by k8s until the last finalizer has been removed.

I could definitely be wrong though :wink:

Reply:
You're right deletion is deferred if the Finalizer is there.
So, prior to that change we could see the [following message](https://github.com/tektoncd/results/pull/703/files#diff-a48f98781edc438b09b76d3a97402d3730a04e55fffb4d049c08545841a4f7afR256) while debugging even if the resource is not deleted properly(due to Finalizer)  which is misleading.